rpc/src/module/chain.rs

```diff
@@ -1645,4 +1692,124 @@ impl ChainRpc for ChainRpcImpl {
         );
         Ok(Some(median_time.into()))
     }
+
+    fn get_median_fee_rate(
+        &self,
+        blocks_to_scan: Option<Uint32>,
+        transactions_to_cal: Option<Uint32>,
+    ) -> Result<ckb_jsonrpc_types::Capacity> {
+        let blocks_to_scan = blocks_to_scan
+            .map(|v| v.value())
+            .unwrap_or(MAX_BLOCKS_TO_SCAN);
+        let transactions_to_cal = transactions_to_cal
+            .map(|v| v.value())
+            .unwrap_or(MIN_TRANSACTIONS);
+
+        // check input parameter
+        if blocks_to_scan > MAX_BLOCKS_TO_SCAN {
+            return Err(RPCError::invalid_params(format!(
+                "invalid input parameter: max_blocks_to_scan, should be less than {}",
+                MAX_BLOCKS_TO_SCAN
+            )));
+        }
+
+        // find matched blocks
+        let snapshot = self.shared.snapshot();
+        let tip = snapshot.tip_header().number();
+        let mut match_blocks = Vec::with_capacity(blocks_to_scan as usize);
+        let mut block_counter = 0;
+        let mut tx_counter = 0;
+
+        // get block hash and check if contains enough transaction
+        let mut hash = snapshot.get_block_hash(tip).ok_or_else(|| {
+            RPCError::custom(RPCError::ChainIndexIsInconsistent, "get_block_hash error")
+        })?;
+        while block_counter < blocks_to_scan {
+            let block = snapshot.get_block(&hash).ok_or_else(|| {
+                RPCError::custom(RPCError::ChainIndexIsInconsistent, "get_block error")
+            })?;
+            block_counter += 1;
+            let block_ext = snapshot.get_block_tx_stat(&hash).ok_or_else(|| {
+                RPCError::custom(
+                    RPCError::ChainIndexIsInconsistent,
+                    "get_block_tx_stat error",
+                )
+            })?;
+
+            // we collect the block hash if it contains more than one transaction(except cellbase),
+            if !block_ext.txs_size.is_empty() {
+                tx_counter += block_ext.txs_size.len();
+                match_blocks.push(hash.clone());
+            }
+
+            if tx_counter >= transactions_to_cal as usize {
+                break;
+            }
+
+            hash = block.parent_hash();
+        }
+
+        if block_counter == blocks_to_scan && tx_counter < transactions_to_cal as usize {
+            return Err(RPCError::invalid_params(format!(
+                "chain hasn't enough transactions in {} recent blocks",
+                blocks_to_scan
+            )));
+        }
+
+        let mut fee_vec: Vec<u64> = Vec::with_capacity(blocks_to_scan as usize);
+        for hash in &match_blocks {
+            let stat = snapshot.get_block_tx_stat(hash).ok_or_else(|| {
+                RPCError::custom(
+                    RPCError::ChainIndexIsInconsistent,
```

Comment:
It's OK that database has no the stat for a block. Choose a new RPC error for this.

```
-1200: ChainBlockTxStatsAreNotFound
```

Reply:
ok, i will modify code as you and quake suggest, thanks!